mica/stats/guide_stats.py

```diff
@@ -0,0 +1,552 @@
+import os
+import time
+import sys
+import argparse
+import numpy as np
+import tables
+from scipy.stats import scoreatpercentile
+from astropy.table import Table
+import logging
+import warnings
+
+from Chandra.Time import DateTime
+import agasc
+from kadi import events
+from Ska.engarchive import fetch, fetch_sci
+import mica.archive.obspar
+from mica.starcheck.starcheck import get_starcheck_catalog_at_date
+import Ska.quatutil
+import Ska.astro
+from mica.quaternion import Quat
+from chandra_aca import dark_model
+
+# Ignore known numexpr.necompiler and table.conditions warning
+warnings.filterwarnings(
+    'ignore',
+    message="using `oa_ndim == 0` when `op_axes` is NULL is deprecated.*",
+    category=DeprecationWarning)
+
+
+logger = logging.getLogger('star_stats')
+logger.setLevel(logging.INFO)
+if not len(logger.handlers):
+    logger.addHandler(logging.StreamHandler())
+
+STAT_VERSION = 0.1
+
+GUIDE_COLS = {
+    'obs': [
+        ('obsid', 'int'),
+        ('obi', 'int'),
+        ('guide_datestart', 'S21'),
+        ('guide_tstart', 'float'),
+        ('kalman_datestop', 'S21'),
+        ('revision', 'S15')],
+    'cat': [
+        ('slot', 'int'),
+        ('idx', 'int'),
+        ('type', 'S5'),
+        ('yang', 'float'),
+        ('zang', 'float'),
+        ('sz', 'S4'),
+        ('mag', 'float')],
+    'stat': [
+        ('n_samples', 'int'),
+        ('not_tracking', 'int'),
+        ('within_0.3', 'int'),
+        ('within_1', 'int'),
+        ('within_3', 'int'),
+        ('within_5', 'int'),
+        ('within_0.3_ok', 'int'),
+        ('within_1_ok', 'int'),
+        ('within_3_ok', 'int'),
+        ('within_5_ok', 'int'),
+        ('within_0.3_modok', 'int'),
+        ('within_1_modok', 'int'),
+        ('within_3_modok', 'int'),
+        ('within_5_modok', 'int'),
+        ('outside_5', 'int'),
+        ('outside_5_ok', 'int'),
+        ('outside_5_modok', 'int'),
+        ('obc_bad_status', 'int'),
+        ('mod_obc_bad_status', 'int'),
+        ('common_col', 'int'),
+        ('quad_bound', 'int'),
+        ('sat_pix', 'int'),
+        ('def_pix', 'int'),
+        ('ion_rad', 'int'),
+        ('mult_star', 'int'),
+        ('aoacmag_min', 'float'),
+        ('aoacmag_mean', 'float'),
+        ('aoacmag_max', 'float'),
+        ('aoacmag_std', 'float'),
+        ('aoacyan_mean', 'float'),
+        ('aoaczan_mean', 'float'),
+        ('dy_min', 'float'),
+        ('dy_mean', 'float'),
+        ('dy_std', 'float'),
+        ('dy_max', 'float'),
+        ('dz_min', 'float'),
+        ('dz_mean', 'float'),
+        ('dz_std', 'float'),
+        ('dz_max', 'float'),
+        ('dr_min', 'float'),
+        ('dr_mean', 'float'),
+        ('dr_std', 'float'),
+        ('dr_5th', 'float'),
+        ('dr_95th', 'float'),
+        ('dr_max', 'float'),
+        ('n_trak_interv', 'int')],
+    'agasc': [
```

Comment:
For the AGASC entry, I would rather see `color` (= `color1`) and `ra`, `dec` being the PM-corrected values.  The rest of the entries from `var` downward are fine.